test/client/erc20.test.mjs

```diff
@@ -0,0 +1,124 @@
+
+// so we can use require with mjs file
+const { expect } = chai;
+chai.use(chaiHttp);
+chai.use(chaiAsPromised);
+
+const environment = config.ENVIRONMENTS[process.env.ENVIRONMENT] || config.ENVIRONMENTS.localhost;
+
+const {
+  fee,
+  transferValue,
+  txPerBlock,
+  tokenConfigs: { tokenType, tokenId },
+  mnemonics,
+  signingKeys,
+} = config.TEST_OPTIONS;
+
+const nf3Users = [new Nf3(signingKeys.user1, environment), new Nf3(signingKeys.user2, environment)];
+
+const web3Client = new Web3Client();
+
+let erc20Address;
+let stateAddress;
+let eventLogs = [];
+
+describe('ERC20 tests', () => {
+  let remainingBalance;
+
+  before(async () => {
+    logger.info('Environment', environment);
+    // console.log('ENV', environment);
+    await nf3Users[0].init(mnemonics.user1);
+    await nf3Users[1].init(mnemonics.user2);
+    erc20Address = '0x499d11e0b6eac7c0593d8fb292dcbbf815fb29ae'; // MATIC
+
+    stateAddress = await nf3Users[0].stateContractAddress;
+    web3Client.subscribeTo('logs', eventLogs, { address: stateAddress });
+  });
+
+  describe('Deposits', () => {
+    it('should increment the balance after deposit some ERC20 crypto', async function () {
+      const before = (await nf3Users[0].getLayer2Balances())[erc20Address]?.[0].balance || 0;
+
+      logger.debug(`Initial L2 balance: ${before}`);
+      // We do txPerBlock deposits of 10 each
+      await depositNTransactions(
+        nf3Users[0],
+        txPerBlock,
+        erc20Address,
+        tokenType,
+        transferValue,
+        tokenId,
+        fee,
+      );
+      eventLogs = await web3Client.waitForEvent(eventLogs, ['blockProposed']);
+      // wait a bit to blockProposed be processed by the client
+      await new Promise(resolving => setTimeout(resolving, 10000));
+      const after = (await nf3Users[0].getLayer2Balances())[erc20Address]?.[0].balance || 0;
+      remainingBalance = txPerBlock * transferValue - (after - before);
+      logger.debug(after, before, remainingBalance);
+      logger.debug(`Remaining L2 balance not in a block yet: ${remainingBalance}`);
+      expect(after).to.be.greaterThan(before);
+    });
+  });
+
+  describe('Transfers', () => {
+    it('should send a single ERC20 transfer directly to a proposer', async function () {
+      const before = (await nf3Users[0].getLayer2Balances())[erc20Address]?.[0].balance || 0;
+
+      const res = await nf3Users[0].transfer(
+        true,
+        erc20Address,
+        tokenType,
+        transferValue,
+        tokenId,
+        nf3Users[1].zkpKeys.compressedPkd,
+        fee,
+      );
+      expect(res).to.be.equal(200);
+
+      const after = (await nf3Users[0].getLayer2Balances())[erc20Address]?.[0].balance || 0;
+      logger.debug(after, before, before + remainingBalance - transferValue);
+      expect(after).to.be.lessThan(before);
+      expect(after).to.satisfy(balance => {
+        return balance < before || balance === before + remainingBalance - transferValue;
```

Comment:
You're relying on that the sequence of the tests will be preserved by the testing library when running them, I'm not sure if this is the best approach, what if someone changes the ordering of the test methods?